### PR TITLE
Fix UART when using RcFast clock source

### PR DESCRIPTION
**Pull request by bugadani**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

#### Description

The clock source was disabled, so after selecting it, nothing happened within the peripheral.

cc #1524

#### Testing

Locally run the HIL test on a C2

### CODE REVIEW

File: hil-test/tests/uart.rs [hunk 1]
```diff
@@ -81,47 +86,23 @@ mod tests {
...
-            // 9600 baud, REF TICK clock source:
-            ctx.uart.change_baud(9600, ClockSource::RefTick);
-            ctx.uart.write(7).ok();
-            let read = block!(ctx.uart.read());
-            assert_eq!(read, Ok(7));
-
-            // 921,600 baud, APB clock source:
-            ctx.uart.change_baud(921_600, ClockSource::Apb);
-            ctx.uart.write(253).ok();
+        let configs = [
```
Comment (MabezDev) [inline hil-test/tests/uart.rs:89]: Nice.

**Review by MabezDev — APPROVED** [review]
LGTM, thanks!

File: hil-test/tests/uart.rs (hunk 1 above)
Comment (bugadani) [reply hil-test/tests/uart.rs:89]: I got a bit fed up with nested `cfg(not)` :D

**Review by jessebraham — APPROVED** [review]
LGTM, thanks!

**Pull request merged**